### PR TITLE
Fix asserts that were always true due to a missed neg.

### PR DESCRIPTION
**Pull request by sandreenko**
It took me a while to understand how in https://github.com/dotnet/runtime/pull/43397#discussion_r504809585 we did not fail on `assert("Unhandled containment for ternary hardware intrinsic with immediate operand");` 

cc @janvorli

### CODE REVIEW

**Comment by sandreenko** [comment]
The assert is failing on `NI_AVX2_ShiftLeftLogical128BitLane`, cc @tannergooding @echesakovMSFT

**Comment by tannergooding** [comment]
Will take a look, it's likely a simple fix.

**Comment by tannergooding** [comment]
@sandreenko, I just added the following at L5467, just above the `default` case for `assert(!"Unhandled containment for binary hardware intrinsic with immediate operand");`:
```cpp
                        case NI_SSE2_ShiftLeftLogical128BitLane:
                        case NI_SSE2_ShiftRightLogical128BitLane:
                        case NI_AVX2_ShiftLeftLogical128BitLane:
                        case NI_AVX2_ShiftRightLogical128BitLane:
                        {
#if DEBUG
                            // These intrinsics should have been marked contained by the general-purpose handling earlier in the method.

                            GenTree* lastOp = HWIntrinsicInfo::lookupLastOp(node);
                            assert(lastOp != nullptr);

                            if (HWIntrinsicInfo::isImmOp(intrinsicId, lastOp) && lastOp->IsCnsIntOrI())
                            {
                                assert(lastOp->isContained());
                            }
#endif

                            break;
                        }
```

**Comment by sandreenko** [comment]
> Do you want me to push a new commit directly to the PR or would you like to just cherry-pick the change (or some variant) yourself?

Please push your changes directly to this PR or you can open a new one and I will close this.

**Review by echesakov — APPROVED** [review]

**Comment by sandreenko** [comment]
Thanks @tannergooding for the fix.

File: src/coreclr/src/jit/lowerxarch.cpp [hunk 1]
```diff
@@ -5464,9 +5464,29 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                             break;
                         }
 
+                        case NI_SSE2_ShiftLeftLogical128BitLane:
+                        case NI_SSE2_ShiftRightLogical128BitLane:
+                        case NI_AVX2_ShiftLeftLogical128BitLane:
+                        case NI_AVX2_ShiftRightLogical128BitLane:
+                        {
+#if DEBUG
+                            // These intrinsics should have been marked contained by the general-purpose handling earlier in the method.
```
Comment (stephentoub) [inline src/coreclr/src/jit/lowerxarch.cpp:5473]: This is causing CI to fail in the formatting job.  It failed on this PR as well.

Comment (tannergooding) [reply src/coreclr/src/jit/lowerxarch.cpp:5473]: Looks like you fixed this in https://github.com/dotnet/runtime/pull/44197, 👍

**Pull request merged**